### PR TITLE
Added doxgen comments : trap.c

### PR DESCRIPTION
**Pull request by kim6515516**
added doxgen comments : trap.c

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

File: hypervisor/trap.c [hunk 1]
```diff
@@ -77,9 +77,12 @@
  For example, copying register values for context switching can be
  performed this way.
  */
-
 static struct arch_regs *_trap_hyp_saved_regs;
-
+/**
+ * @brief Handler for data abort exception
```
Comment (Wonseok) [inline hypervisor/trap.c:82]: s/Handler for data abort exception/Handles data abort exception taken from a mode other than Hyp mode

File: hypervisor/trap.c [hunk 2]
```diff
@@ -77,9 +77,12 @@
...
+ * @param Current register's value of general purpose, program counter, lr, cpsr
```
Comment (Wonseok) [inline hypervisor/trap.c:83]: Please do not use work with 's.

Comment (Wonseok) [inline hypervisor/trap.c:83]: Too long description.
ARM registers which includes 13 general purpose register r0-r12, 1 Stack Pointer (SP), 1 Link Register (LR), 1 Program Counter (PC).

Comment (Wonseok) [inline hypervisor/trap.c:83]: Please explain why do we need to use input parameters such as regs?

File: hypervisor/trap.c [hunk 3]
```diff
@@ -77,9 +77,12 @@
...
+ * @return Status for Hypervisor
```
Comment (Wonseok) [inline hypervisor/trap.c:84]: What is the status? What is the difference of return type?

**Pull request closed without merging**